shell.nix

```diff
@@ -6,10 +6,11 @@
 with import <nixpkgs> {};
 with import (fetchTarball "https://github.com/nix-community/nixGL/archive/489d6b095ab9d289fe11af0219a9ff00fe87c7c5.tar.gz") { enable32bits = false; };
 let
-	llvmPackages = llvmPackages_14;
-	stdenv = stdenvAdapters.useMoldLinker clangStdenv;
+	pkgs_gnumake_4_3 = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/6adf48f53d819a7b6e15672817fa1e78e5f4e84f.tar.gz") {};
+	llvmPackages = llvmPackages_14; # servo/servo#31059
+	stdenv = stdenvAdapters.useMoldLinker llvmPackages.stdenv;
 in
-	stdenv.mkDerivation {
+	llvmPackages.stdenv.mkDerivation {
```

Comment:
Thanks very much for your PR and for tracking down all these upstream issues :tada: 

The changes look mostly good to me, but is there a reason you resort back to `llvmPackages.stdenv` here instead of using just `stdenv`, which was defined as `stdenvAdapters.useMoldLinker llvmPackages.stdenv`, i.e. you are effectively removing mold as linker here?

Reply:
thanks! this was a mistake, i intended to prepend it to the stdenv in the definition only.

Reply:
fixed in 64419d9909837bcad5b2b626ac74f1134dffc137